game-core/src/main/java/games/strategy/engine/framework/ui/SaveGameFileChooser.java

```diff
@@ -8,25 +8,41 @@
 import javax.swing.JFileChooser;
 import javax.swing.filechooser.FileFilter;
 
+import com.google.common.annotations.VisibleForTesting;
+
 import games.strategy.engine.framework.GameDataFileUtils;
-import games.strategy.engine.framework.headlessGameServer.HeadlessGameServer;
 import games.strategy.triplea.settings.ClientSetting;
 
-public class SaveGameFileChooser extends JFileChooser {
+/**
+ * A file chooser for save games. Defaults to the user's configured save game folder.
+ *
+ * <p>
+ * Also provides several methods for getting the names of auto-save files periodically generated during a game.
+ * </p>
+ */
+public final class SaveGameFileChooser extends JFileChooser {
   private static final long serialVersionUID = 1548668790891292106L;
-  private static final String AUTOSAVE_FILE_NAME = GameDataFileUtils.addExtension("autosave");
-  private static final String AUTOSAVE_ODD_ROUND_FILE_NAME = GameDataFileUtils.addExtension("autosave_round_odd");
-  private static final String AUTOSAVE_EVEN_ROUND_FILE_NAME = GameDataFileUtils.addExtension("autosave_round_even");
+
+  @VisibleForTesting
+  static final String HEADLESS_AUTOSAVE_FILE_NAME = GameDataFileUtils.addExtension("autosave");
+  @VisibleForTesting
+  static final String ODD_ROUND_AUTOSAVE_FILE_NAME = GameDataFileUtils.addExtension("autosave_round_odd");
+  @VisibleForTesting
+  static final String EVEN_ROUND_AUTOSAVE_FILE_NAME = GameDataFileUtils.addExtension("autosave_round_even");
+
   private static SaveGameFileChooser instance;
```

Comment:
Note, that this is unrelated to the PR, just some general information I'd like to share:

This seems like a code smell to me, there's absolutely no good reason (I could think of) to hold on to a swing object, via a static reference, that's obviously intended for one-time use only.
I looked through the usages of the getInstance methods, and they are really just using this method to display either a save or an open dialog.
Theres a lot of code going through GameRunner to set the mainFrame as parent component which is clearly not used for this dialog, only for the classic AWT dialog for mac users.
I have access to a mac at work, so I might be able to check if the thing with the AWT dialog vs Swing dialog still holds true nowadays.

TL;DR: Mac and other OSes behaviour should be unified (parent component should be mainFrame or something) and a SaveGameFileChooser instance should be created dynamically, not as a sort of singleton.

Reply:
Yup, I thought the same thing. :smile: